src/xdist/workermanage.py

```diff
@@ -312,6 +312,11 @@ def shutting_down(self) -> bool:
 
     def setup(self) -> None:
         self.log("setting up worker session")
+        # Cache rinfo for backward compatibility, since pytest-cov
+        # accesses rinfo while the main thread is busy executing our
+        # remote_exec call, which triggers a deadlock error for the
+        # main_thread_only execmodel if the rinfo has not been cached.
+        self.gateway._rinfo()
```

Comment:
> This change LGTM, pending [pytest-dev/execnet#274 (comment)](https://github.com/pytest-dev/execnet/pull/274#issuecomment-2080397459) (and if we decide to fix in xdist instead of execnet, should update the "Remove this after execnet is fixed" comment)

Okay, comment updated.